### ISSUE TITLE
Unable to convert Qwen2.5 to mediapipe supported format

### ISSUE THREAD

**Issue by atultiwari** [google-ai-edge/litert-torch#275]
### Description of the bug:

I used [AI-Edge-Torch](https://github.com/google-ai-edge/ai-edge-torch.git) to convert `Qwen/Qwen2.5-0.5B-Instruct` from Huggingface to `tflite`, but as decribed in the [documentation](https://ai.google.dev/edge/mediapipe/solutions/genai/llm_inference#ai_edge_model_conversion), I am unable to proceed to the next step. As per documentation -
1. LLM mapped to a TFLite model through AI Edge - Done & created a file named `qwen_0_5b_q8_seq1024_ekv1280.tflite`
2. **Create a Task Bundle - I am facing issues here.**

The second step provides the following sample code snippet -
````
import mediapipe as mp
from mediapipe.tasks.python.genai import bundler

config = bundler.BundleConfig(
    tflite_model=TFLITE_MODEL,
    tokenizer_model=TOKENIZER_MODEL,
    start_token=START_TOKEN,
    stop_tokens=STOP_TOKENS,
    output_filename=OUTPUT_FILENAME,
    enable_bytes_to_unicode_mapping=ENABLE_BYTES_TO_UNICODE_MAPPING,
)
bundler.create_bundle(config)
````

**But the problem is - [Qwen2.5](https://huggingface.co/Qwen/Qwen2.5-0.5B-Instruct/tree/main) doesn't provide Tokenizer model file. How can I make it?**
I also wanted to know if I need to put the output file name something like `qwen.task`. If yes then what would be the next step, it's also unclear to me.
**Basically I am unable to understand, how I can use this generated file in the mediapipe sample app?**

If needed - colab notebook [link](https://colab.research.google.com/drive/13H9Xs6Jen_zd3NI0QnOWL7kb5bM5RupO?usp=sharing).

Thank you.
Regards
Atul

### Any other information you'd like to share?

I originally asked the same question `mediapipe-samples repo (question [link](https://github.com/google-ai-edge/mediapipe-samples/issues/468)), but there I was suggested to ask it here.

**Comment by pkgoogle** [comment]
Hi @atultiwari, can you use the tokenizer object from the quickstart from that model? https://huggingface.co/Qwen/Qwen2.5-0.5B-Instruct#quickstart

**Comment by atultiwari** [comment]
> Hi @atultiwari, can you use the tokenizer object from the quickstart from that model? https://huggingface.co/Qwen/Qwen2.5-0.5B-Instruct#quickstart

Hi, I have already tried that in the already shared colab notebook, but I forgot to describe here.
**When I used tokennizer object from the quickstart, I encountered following error -**
```
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
[<ipython-input-4-3e13d56bb80a>](https://localhost:8080/#) in <cell line: 12>()
     10     enable_bytes_to_unicode_mapping=ENABLE_BYTES_TO_UNICODE_MAPPING,
     11 )
---> 12 bundler.create_bundle(config)

[/usr/local/lib/python3.10/dist-packages/mediapipe/tasks/python/genai/bundler/llm_bundler.py](https://localhost:8080/#) in create_bundle(config)
     73     artifacts[_BundleTags.TF_LITE_PREFILL_DECODE.name] = f.read()
     74 
---> 75   with open(config.tokenizer_model, "rb") as f:
     76     artifacts[_BundleTags.TOKENIZER_MODEL.name] = f.read()
     77 

TypeError: expected str, bytes or os.PathLike object, not Qwen2TokenizerFast
```

This is how I tried it -

```
from transformers import AutoTokenizer, AutoModelForCausalLM
TOKENIZER = AutoTokenizer.from_pretrained("Qwen/Qwen2.5-0.5B-Instruct")

TFLITE_MODEL = "/content/drive/MyDrive/AI/ai_edge_torch/converted_models/qwen_0_5b_q8_seq1024_ekv1280.tflite"
TOKENIZER_MODEL = TOKENIZER
START_TOKEN = "<|im_start|>"
STOP_TOKENS = ["<|endoftext|>"]
OUTPUT_FILENAME = "qwen05.bin"
ENABLE_BYTES_TO_UNICODE_MAPPING = True

import mediapipe as mp
from mediapipe.tasks.python.genai import bundler

config = bundler.BundleConfig(
    tflite_model=TFLITE_MODEL,
    tokenizer_model=TOKENIZER,
    start_token=START_TOKEN,
    stop_tokens=STOP_TOKENS,
    output_filename=OUTPUT_FILENAME,
    enable_bytes_to_unicode_mapping=ENABLE_BYTES_TO_UNICODE_MAPPING,
)
bundler.create_bundle(config)
```

**Comment by talumbau** [comment]
Yes, `TOKENIZER` above should be a path to an actual file that encodes the tokenizer for the model. We have some documentation on this here:

https://ai.google.dev/edge/mediapipe/solutions/genai/llm_inference/#convert-model

but it could be better. The typical input would be something like this `.model` file:

https://github.com/google/gemma_pytorch/blob/main/tokenizer/tokenizer.model

**Comment by atultiwari** [comment]
> Yes, `TOKENIZER` above should be a path to an actual file that encodes the tokenizer for the model. We have some documentation on this here:
> 
> https://ai.google.dev/edge/mediapipe/solutions/genai/llm_inference/#convert-model
> 
> but it could be better. The typical input would be something like this `.model` file:
> 
> https://github.com/google/gemma_pytorch/blob/main/tokenizer/tokenizer.model

I have checked the mentioned documentation, that's why I posted this issue.
So, it seems a dead end for now.

**Comment by talumbau** [comment]
Ah @atultiwari now I understand. Sorry for the confusion. Yes, the current API requires a `.model` file for the tokenizer from the [SentencePiece](https://github.com/google/sentencepiece/blob/master/python/README.md) framework. So the task here would be to create a SentencePiece model from the `tokenizer.json`, `tokenizer_config.json` and `vocab.json` files provided on the HF page for Qwen 2.5. If one does not already exist somewhere, you can use the framework to make such a `.model` file. We should make this workflow easier but currently this is a gap and users will have to create their own `.model` files from SentencePiece of one is not provided with your chosen model.

**Comment by atultiwari** [comment]
Thank you @talumbau for your reply. I understood the gap in the steps and checked out `SentencePiece` framework too. Unfortunately I am not able to figure out how to create a `.model` file from the `tokenizer.json`, `tokenizer_config.json` and `vocab.json` using this framework.

This is what I understood, that how should I use this framework, but I am unable to figure out where should I mention the specific `json` files -
```
pip install sentencepiece
import sentencepiece as spm
spm.SentencePieceTrainer.train(input='data.txt', model_prefix='qwen', vocab_size=5000)
```
I am sorry, if my question is very basic, but I tried my best to figure out myself, but I couldn't.

**Comment by godguy23** [comment]
@atultiwari Hi, Did you solved this problem now ? Can you show me the code which to convert the qwen to a tflite files?

**Comment by haooxia** [comment]
@atultiwari Hi, do you know how to create a .model file from tokenizer.json, tokenizer_config.json, and vocab.json using this framework now?

I got the Qwen3 `.tflite` model, but without the `tokenizer.model`, it seems I can't make any progress.

**Comment by aravindmurali22** [comment]
Hello @atultiwari , Since this issue was opened, significant updates and improvements have been released that may have already addressed your concern. We are closing this issue to maintain a current and relevant backlog. Please test using the latest version of the library. If the issue persists, kindly open a new report with updated reproduction steps so we can investigate it within our current development workflow.
Thank you.
